### PR TITLE
add metadata manager to shared package

### PR DESCRIPTION
**Pull request by lkostrowski**
## Scope of the PR

1. Extract metadata-manager (copy-pasted in all apps) to shared package
2. Used this factory in cms-2 and segment apps (no-config change)

Other apps must be updated in further PRs, due to some refactor needed (e.g. appId must be present)

## Checklist

- [ ] `.github/dependabot.yaml` is up-to date.
- [ ] I added changesets and [read good practices](/.changeset/README.md).

### CODE REVIEW

**Comment by lkostrowski** [comment]
@krzysztofwolski @peelar please handle this update in apps you maintain

File: apps/invoices/src/modules/app-configuration/__mocks__/metadata-manager.ts [hunk 1]
```diff
@@ -6,6 +6,7 @@ import { SimpleGraphqlClient } from "../metadata-manager";
 export const mockMetadataManager = {
   set: vi.fn().mockImplementation(async () => {}),
   get: vi.fn().mockImplementation(async () => {}),
+  delete: vi.fn().mockImplementation(async () => {}),
```
Comment (lkostrowski) [inline apps/invoices/src/modules/app-configuration/__mocks__/metadata-manager.ts:9]: adding missing types, not related to the scope

File: apps/search/src/test/handlers/webhooks-status-handler.test.ts [hunk 2]
```diff
@@ -62,6 +64,7 @@ describe("webhooksStatusHandler", () => {
   const settingsManagerMock: SettingsManager = {
     get: vi.fn(),
     set: vi.fn(),
+    delete: vi.fn(),
```
Comment (lkostrowski) [inline apps/search/src/test/handlers/webhooks-status-handler.test.ts:67]: adding missing types, not related to the scope

File: packages/shared/package.json [hunk 3]
```diff
@@ -27,6 +25,8 @@
     "next": "13.4.8",
     "react": "18.2.0",
     "react-dom": "18.2.0",
+    "typescript": "5.1.6",
+    "urql": "^4.0.4",
```
Comment (lkostrowski) [inline packages/shared/package.json:29]: moved to dev deps and peer - this package should not force any dependencies

File: packages/shared/src/metadata-manager.ts [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+    item?: {
+      __typename?: "App";
+      privateMetadata: Array<{ __typename?: "MetadataItem"; key: string; value: string }>;
+    } | null;
+  } | null;
+};
+
+export type MetadataManagerGraphqlClient = Pick<Client, "mutation" | "query">;
+
+async function fetchAllPrivateMetadata(
```
Comment (krzysztofwolski) [inline packages/shared/src/metadata-manager.ts:69]: **Question[^1][^2]**:
What do you think about exporting this function too? I'm using these queries not only in App metadata, but for Category too (products feed app)

[^1]: ✅ Not blocking the review. 
[^2]: I just wanted to know more context, or spotted a cool solution I would like to know more about :)

**Review by krzysztofwolski — APPROVED** [review]
Great, really like the idea to remove the duplication.
We can create the issue to use it in the other apps too.

File: packages/shared/src/metadata-manager.ts (hunk 4 above)
Comment (lkostrowski) [reply packages/shared/src/metadata-manager.ts:69]: I dont tthink we should mix it here. I mean, I will export it and you will create a coupling between metadata manager and categories, which was not meant to happen.

I think its ok to extract "common" queries to shared, but then lets create a separate package for it and explicitly design it to work with categories etc (or anything)

**Pull request merged**